src/bindings/jna/libelektra/src/main/java/org/libelektra/Elektra.java

```diff
@@ -227,64 +227,7 @@ int elektraGOptsContractFromStrings (Pointer contractKeySet, long argsSize, Stri
 
 	// Key methods --------------------------------------------------------------
 
-	/**
-	 * Enumeration of argument flags for {@link Elektra#keyNew(String, Object...)}.
-	 */
-	enum KeyNewArgumentFlags
-	{
-
-		/**
-		 * Used as a parameter terminator to {@link Elektra#keyNew(String, Object...)}
-		 */
-		KEY_END (0),
-
-		/**
-		 * Flag for the key name
-		 */
-		KEY_NAME (1),
-
-		/**
-		 * Flag for the key data
-		 */
-		KEY_VALUE (1 << 1),
-
-		/**
-		 * Flag for the key comment
-		 */
-		KEY_COMMENT (1 << 3),
-
-		/**
-		 * Flag if the key is binary
-		 */
-		KEY_BINARY (1 << 4),
-
-		/**
-		 * Flag for maximum size to limit value
-		 */
-		KEY_SIZE (1 << 11),
-
-		/**
-		 * Flag for metadata
-		 */
-		KEY_META (1 << 15);
-
-		private final Integer value;
-
-		/**
-		 * @return Integer value recognized by the native library.
-		 */
-		public Integer getValue ()
-		{
-			return value;
-		}
-
-		private KeyNewArgumentFlags (int value)
-		{
-			this.value = Integer.valueOf (value);
-		}
-	}
-
-	public static final String KEY_LOCAL_NAME = "/";
+	static final String KEY_LOCAL_NAME = "/";
```

Comment:
I don't think it is a good idea to introduce such a "LOCAL_NAME", it introduces very different vocabulary than Elektra usually uses. If you need/want such aliases, introduce one for each namespace with the name of the namespace.

Reply:
The purpose of this constant is to enable creating local key object, not storable in key database, for use in `errorKey` style API. I concur, that the API should be more concise ant therefore removed the constant and introduced `Key::createNameless`. I need to pass "/" instead of `NULL` because the `keyNew` documentation is wrong:
https://github.com/ElektraInitiative/libelektra/blob/9471062a048e77024e3b3b934bc3a944bb4b5ab7/src/libs/elektra/key.c#L140-L152

Reply:
Additionally we might ask the question whether to completely remove the possibility to specify an errorKey in the Java API, since we always wand to convert it to an exception anyways... and therefore could always implicitly create a temporary nameless key for that purpose

Reply:
Please let me know what you think about it so i can create an appropriate issue

Reply:
Probably I need more insight about the error concept in the binding. I think first step would be to describe the error concept somewhere. I agree it is not relevant for this release.

Reply:
see #3873